### PR TITLE
Maven compiler properties panel updates

### PR DESCRIPTION
**Pull request by mbien**
Maven `CompilePanel` improvements

 - move language level combo box from source to compile panel, to be next to the JDK picker and update UI
 - sort JDK list and lang level list in reverse
 - lang level list should display the correct range for the selected JDK version and validate selection
 - ModelUtils.setSourceLevel should check plugin management too
 - many bugfixes

Maven `CompilePanel`: implement `-Xlint` checkbox

 - replaces the show-deprecations check box with the more powerful javac lints (includes deprecation hint too)
 - lint flags are automatically picked up by the editor as hints so we get this integration for free
 
 screenshot:
 image

### CODE REVIEW

**Review by matthiasblaesing — COMMENTED** [review]
Found a stray `System.out` and from a quick test + glance this looks ok.

I'm not happy with the Reflection of default values in the checkboxes. For me the bold formatting looked random and only with the default in mind, I can deduce the meaning. 

I see a dropdown as an option with value: Unset (Default: <whatever>)/Enabled/Disabled. Then people don't need to guess as options are obvious.

File: java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java [hunk 1]
```diff
@@ -218,63 +235,86 @@ public boolean getDefaultValue() {
...
+
+            @Override
+            public void setValue(Boolean value) {
+                if (operation != null) {
+                    handle.removePOMModification(operation);
+                }
+                if (value == null) {
+                    return;
+                }
+                System.out.println(value);
```
[the remark this answers is not in the document]
Comment (mbien) [reply java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java:295]: oops, thanks!

[the remark this answers is not in the document]
Comment (mbien) [reply java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java:295]: Regarding the visualization of the inherited state. This was not my idea, I only fixed it. This does work for more components than just the check box + it adds a tool tip. 

So its more complicated than a 3 state check box. I am not against taking another look at it at some point but this was not the goal of this PR originally. 

This was about bugs and UX. I didn't want to have an invalid source level every time I switched the JDK during testing - this became a larger PR than I anticipated already + it spawned 2 more PRs during tests.

**Review by matthiasblaesing — APPROVED** [review]
The checkbox implementation is something I'd like to talk about with the original developer. It falls into the category of things, that seem to be to elaborate and broken at the same time as that could be coincidence. Anyway, if it was there, lets keep it.

File: java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbien) [reply java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java]: the "updater" API is here btw: https://github.com/apache/netbeans/tree/master/java/maven/src/org/netbeans/modules/maven/api/customizer/support

**Comment by jglick** [comment]
The width on the new dropdown is not working for me with `-fontsize 24`. 
Screenshot from 2024-10-03 16-26-39
(Here 17 is selected, but you can only see that by clicking the dropdown.) Maybe it should be allowed to float to the dialog width as well?

**Comment by mbien** [comment]
@jglick I take a look. This usually happens when a component is set to a fixed size by accident instead of leaving it at "default" which would pack it automatically.

**Comment by mbien** [comment]
https://github.com/apache/netbeans/pull/7830 should fix it. PR will produce a dev build in a few minutes (see workflow summary).

**Pull request merged**